admin-dev/themes/new-theme/js/pages/product/components/combination-list/bulk-choices-selector.ts

```diff
@@ -40,16 +41,43 @@ export default class BulkChoicesSelector {
 
   private tabContainer: HTMLDivElement;
 
-  constructor(eventEmitter: EventEmitter, tabContainer: HTMLDivElement) {
+  private paginatedCombinationsService: PaginatedCombinationsService
+
+  constructor(
+    eventEmitter: EventEmitter,
+    tabContainer: HTMLDivElement,
+    paginatedCombinationsService: PaginatedCombinationsService,
+  ) {
     this.eventEmitter = eventEmitter;
     this.tabContainer = tabContainer;
+    this.paginatedCombinationsService = paginatedCombinationsService;
     this.init();
   }
 
+  //@todo: may become private or unused
   public getSelectedCheckboxes(): NodeListOf<HTMLInputElement> {
     return this.tabContainer.querySelectorAll<HTMLInputElement>(`${CombinationMap.tableRow.isSelectedCombination}:checked`);
   }
 
+  //@todo: could as well return count?
+  public async getSelectedIds(): Promise<number[]> {
```

Comment:
I'm not sure about the count If it's the count of selected IDs you can just use `selectedIds.length`
If it's about the total number of combinations in the list (with or without filters) I believe this can be fetched from the paginator which has this data

Although it may be useful to have a procy method in this class lie `getTotal() { return paginator.total; }` if it's needed outside of this component. But although I'm pretty sur this total will be needed internally to update the checkbox labels, I'm not sure that it needs to expose it via a public method.